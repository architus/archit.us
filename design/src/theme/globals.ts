import { css } from "linaria";

import { injectColorGlobals, color } from "./color";
import { injectMixinGlobals } from "./mixins";
import { injectTypographyGlobals } from "./typography";

// From https://hankchizljaw.com/wrote/a-modern-css-reset/
export const globals = css`
  :global() {
    ${injectColorGlobals()}
    ${injectTypographyGlobals()}
    ${injectMixinGlobals()}

    // Box sizing rules
    *,
    *::before,
    *::after {
      box-sizing: border-box;
    }

    // Remove default padding
    ul[class],
    ol[class] {
      padding: 0;
    }

    // Remove default margin
    body,
    h1,
    h2,
    h3,
    h4,
    h5,
    h6,
    p,
    ul,
    ol,
    li,
    figure,
    figcaption,
    blockquote,
    dl,
    dd {
      margin: 0;
    }

    // Set core body defaults
    body {
      margin: 0;
      min-height: 100vh;
      scroll-behavior: smooth;
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
      text-rendering: optimizeSpeed;

      line-height: 1.5;
      color: ${color("text")};
      background-color: ${color("bg")};
    }

    // Remove list styles on ul, ol elements with a class attribute
    ul[class],
    ol[class] {
      list-style: none;
    }

    // Inherit fonts for inputs and buttons
    input,
    button,
    textarea,
    select {
      font: inherit;
    }

    // Remove all animations and transitions for people that prefer not to see them
    @media (prefers-reduced-motion: reduce) {
      * {
        animation-duration: 0.01ms !important;
        animation-iteration-count: 1 !important;
        transition-duration: 0.01ms !important;
        scroll-behavior: auto !important;
      }
    }
  }
`;
