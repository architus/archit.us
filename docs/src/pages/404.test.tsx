import { jest, describe, it, expect } from "@jest/globals";
import React from "react";
import renderer from "react-test-renderer";

import "@docs/test/watch-media";
import NotFoundPage from "./404";

jest.mock("@docs/data/build-metadata");
jest.mock("@docs/data/navigation-tree");
jest.mock("@docs/data/seo-data");
jest.mock("@docs/data/site-title");
jest.mock("@docs/data/socials");
jest.mock("@docs/data/path-prefix");
jest.mock("@docs/data/footer-data");

describe("<404 />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<NotFoundPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
