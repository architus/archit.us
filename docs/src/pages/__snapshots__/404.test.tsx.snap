// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<404 /> renders correctly 1`] = `
Array [
  <nav
    className="Header_h63g0gq"
    mode="dark"
    style={
      Object {
        "--h63g0gq-10": "var(--c-primary)",
        "--h63g0gq-9": "8px",
      }
    }
  >
    <a
      className="LogoLink_lwue83p"
      href="/"
    >
      <div
        className="Brand_biutx5f"
        style={
          Object {
            "--biutx5f-0": "0",
          }
        }
      >
        <svg
          className="Logo_l81tux5"
          height={36}
        >
          __svg__
        </svg>
        <h1
          className="SiteTitle_s16u4h4e"
        >
          Documentation
        </h1>
      </div>
    </a>
    <div
      className="HeaderLinksWrapper_hfrfs33"
    />
    <div
      className="RightComponents_r4rjoor"
    >
      <button
        className="buttonClass_b1jnjt8r"
        onClick={[Function]}
        title="Switch color mode"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M14.53 10.53a7 7 0 01-9.058-9.058A7.003 7.003 0 008 15a7.002 7.002 0 006.53-4.47z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
  </nav>,
  <div
    className="Layout_llw1ymp"
    style={
      Object {
        "--llw1ymp-2": "none",
        "--llw1ymp-3": "0",
        "--llw1ymp-5": "translateX(-100%)",
      }
    }
  >
    <div
      className="Drawer_d13sumzj"
    >
      <nav
        className="SideNav_s3xpfh4"
      >
        <h3
          className="SideNavHeader_s1kyk6un"
          style={
            Object {
              "--s1kyk6un-4": "0",
            }
          }
        >
          Unknown
        </h3>
      </nav>
    </div>
    <div
      className="DrawerExpander_dop0li9"
    >
      <button
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </button>
    </div>
    <div
      className="DrawerOverlay_dcd8904"
      onClick={[Function]}
    />
    <main
      className="Content_chf6d6q"
    >
      <div
        className="ContentExpand_cwcyb8o"
      >
        <div
          className="Outer_o14wiqln"
        >
          <h1
            className="Title_tj6xfbl"
          >
            Page Not Found
          </h1>
          <article
            className="Article_a19ycgqa"
          >
            <h4
              className="StatusCode_s1vzh75s"
            >
              Status code 404
            </h4>
            <p>
              The page you're looking for doesn't exist or was removed.
            </p>
            <p>
              If you feel this is an error, please file a new issue
               
              <a
                className="baseLinkClass_b190u9vd"
                href="https://github.com/architus/archit.us/issues/new"
                rel="noreferrer noopener"
                target="_blank"
              >
                on GitHub
                <span
                  className="ExternalIcon_eixdjzj"
                  style={
                    Object {
                      "--eixdjzj-0": "0",
                      "--eixdjzj-1": "8px",
                    }
                  }
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g>
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M10 6v2H5v11h11v-5h2v6a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1V7a1 1 0 0 1 1-1h6zm11-3v8h-2V6.413l-7.793 7.794-1.414-1.414L17.585 5H13V3h8z"
                      />
                    </g>
                  </svg>
                </span>
              </a>
            </p>
          </article>
        </div>
      </div>
      <footer
        className="Footer_fnkqr06"
      >
        <div
          className="PrimaryContent_p15lomux FooterContent_f1iwiaxb"
        >
          <div>
            <h3
              className="FooterHeader_ff6xenz"
            >
              About
            </h3>
            <hr
              className="FooterDivider_f112mulj"
            />
            __about__
          </div>
          <div>
            <h3
              className="FooterHeader_ff6xenz"
            >
              Links
            </h3>
            <hr
              className="FooterDivider_f112mulj"
            />
          </div>
          <div>
            <aside
              className="BrandCard_b1ln16bc Card_c3vzevn"
            >
              <div
                className="Brand_bw8neyp"
              >
                <div
                  className="Brand_biutx5f"
                  style={
                    Object {
                      "--biutx5f-0": "13px",
                    }
                  }
                >
                  <svg
                    className="Logo_l81tux5"
                    height={36}
                  >
                    __svg__
                  </svg>
                  <h1
                    className="SiteTitle_s16u4h4e"
                  >
                    Documentation
                  </h1>
                </div>
              </div>
            </aside>
          </div>
        </div>
      </footer>
      <footer
        className="SecondaryFooter_s11awdkr"
      >
        <div
          className="SecondaryContent_suumtrm FooterContent_f1iwiaxb"
        >
          Built with 
          <svg
            className="HeartIcon_h1ia68hq"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
           using 
          <div
            className="TechnologyList_t1t472gb"
          >
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <a
                className="TechnologyLink_t1j6hkq4 baseLinkClass_b190u9vd"
                href="https://github.com/architus/archit.us"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 496 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  />
                </svg>
              </a>
            </span>
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <a
                className="TechnologyLink_t1j6hkq4 baseLinkClass_b190u9vd"
                href="https://reactjs.org/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M418.2 177.2c-5.4-1.8-10.8-3.5-16.2-5.1.9-3.7 1.7-7.4 2.5-11.1 12.3-59.6 4.2-107.5-23.1-123.3-26.3-15.1-69.2.6-112.6 38.4-4.3 3.7-8.5 7.6-12.5 11.5-2.7-2.6-5.5-5.2-8.3-7.7-45.5-40.4-91.1-57.4-118.4-41.5-26.2 15.2-34 60.3-23 116.7 1.1 5.6 2.3 11.1 3.7 16.7-6.4 1.8-12.7 3.8-18.6 5.9C38.3 196.2 0 225.4 0 255.6c0 31.2 40.8 62.5 96.3 81.5 4.5 1.5 9 3 13.6 4.3-1.5 6-2.8 11.9-4 18-10.5 55.5-2.3 99.5 23.9 114.6 27 15.6 72.4-.4 116.6-39.1 3.5-3.1 7-6.3 10.5-9.7 4.4 4.3 9 8.4 13.6 12.4 42.8 36.8 85.1 51.7 111.2 36.6 27-15.6 35.8-62.9 24.4-120.5-.9-4.4-1.9-8.9-3-13.5 3.2-.9 6.3-1.9 9.4-2.9 57.7-19.1 99.5-50 99.5-81.7 0-30.3-39.4-59.7-93.8-78.4zM282.9 92.3c37.2-32.4 71.9-45.1 87.7-36 16.9 9.7 23.4 48.9 12.8 100.4-.7 3.4-1.4 6.7-2.3 10-22.2-5-44.7-8.6-67.3-10.6-13-18.6-27.2-36.4-42.6-53.1 3.9-3.7 7.7-7.2 11.7-10.7zM167.2 307.5c5.1 8.7 10.3 17.4 15.8 25.9-15.6-1.7-31.1-4.2-46.4-7.5 4.4-14.4 9.9-29.3 16.3-44.5 4.6 8.8 9.3 17.5 14.3 26.1zm-30.3-120.3c14.4-3.2 29.7-5.8 45.6-7.8-5.3 8.3-10.5 16.8-15.4 25.4-4.9 8.5-9.7 17.2-14.2 26-6.3-14.9-11.6-29.5-16-43.6zm27.4 68.9c6.6-13.8 13.8-27.3 21.4-40.6s15.8-26.2 24.4-38.9c15-1.1 30.3-1.7 45.9-1.7s31 .6 45.9 1.7c8.5 12.6 16.6 25.5 24.3 38.7s14.9 26.7 21.7 40.4c-6.7 13.8-13.9 27.4-21.6 40.8-7.6 13.3-15.7 26.2-24.2 39-14.9 1.1-30.4 1.6-46.1 1.6s-30.9-.5-45.6-1.4c-8.7-12.7-16.9-25.7-24.6-39s-14.8-26.8-21.5-40.6zm180.6 51.2c5.1-8.8 9.9-17.7 14.6-26.7 6.4 14.5 12 29.2 16.9 44.3-15.5 3.5-31.2 6.2-47 8 5.4-8.4 10.5-17 15.5-25.6zm14.4-76.5c-4.7-8.8-9.5-17.6-14.5-26.2-4.9-8.5-10-16.9-15.3-25.2 16.1 2 31.5 4.7 45.9 8-4.6 14.8-10 29.2-16.1 43.4zM256.2 118.3c10.5 11.4 20.4 23.4 29.6 35.8-19.8-.9-39.7-.9-59.5 0 9.8-12.9 19.9-24.9 29.9-35.8zM140.2 57c16.8-9.8 54.1 4.2 93.4 39 2.5 2.2 5 4.6 7.6 7-15.5 16.7-29.8 34.5-42.9 53.1-22.6 2-45 5.5-67.2 10.4-1.3-5.1-2.4-10.3-3.5-15.5-9.4-48.4-3.2-84.9 12.6-94zm-24.5 263.6c-4.2-1.2-8.3-2.5-12.4-3.9-21.3-6.7-45.5-17.3-63-31.2-10.1-7-16.9-17.8-18.8-29.9 0-18.3 31.6-41.7 77.2-57.6 5.7-2 11.5-3.8 17.3-5.5 6.8 21.7 15 43 24.5 63.6-9.6 20.9-17.9 42.5-24.8 64.5zm116.6 98c-16.5 15.1-35.6 27.1-56.4 35.3-11.1 5.3-23.9 5.8-35.3 1.3-15.9-9.2-22.5-44.5-13.5-92 1.1-5.6 2.3-11.2 3.7-16.7 22.4 4.8 45 8.1 67.9 9.8 13.2 18.7 27.7 36.6 43.2 53.4-3.2 3.1-6.4 6.1-9.6 8.9zm24.5-24.3c-10.2-11-20.4-23.2-30.3-36.3 9.6.4 19.5.6 29.5.6 10.3 0 20.4-.2 30.4-.7-9.2 12.7-19.1 24.8-29.6 36.4zm130.7 30c-.9 12.2-6.9 23.6-16.5 31.3-15.9 9.2-49.8-2.8-86.4-34.2-4.2-3.6-8.4-7.5-12.7-11.5 15.3-16.9 29.4-34.8 42.2-53.6 22.9-1.9 45.7-5.4 68.2-10.5 1 4.1 1.9 8.2 2.7 12.2 4.9 21.6 5.7 44.1 2.5 66.3zm18.2-107.5c-2.8.9-5.6 1.8-8.5 2.6-7-21.8-15.6-43.1-25.5-63.8 9.6-20.4 17.7-41.4 24.5-62.9 5.2 1.5 10.2 3.1 15 4.7 46.6 16 79.3 39.8 79.3 58 0 19.6-34.9 44.9-84.8 61.4zm-149.7-15c25.3 0 45.8-20.5 45.8-45.8s-20.5-45.8-45.8-45.8c-25.3 0-45.8 20.5-45.8 45.8s20.5 45.8 45.8 45.8z"
                  />
                </svg>
              </a>
            </span>
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <a
                className="TechnologyLink_t1j6hkq4 baseLinkClass_b190u9vd"
                href="https://www.gatsbyjs.org/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg>
                  __svg__
                </svg>
              </a>
            </span>
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <a
                className="TechnologyLink_t1j6hkq4 baseLinkClass_b190u9vd"
                href="https://linaria.now.sh/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg>
                  __svg__
                </svg>
              </a>
            </span>
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <a
                className="TechnologyLink_t1j6hkq4 baseLinkClass_b190u9vd"
                href="https://graphql.org/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg>
                  __svg__
                </svg>
              </a>
            </span>
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <a
                className="TechnologyLink_t1j6hkq4 baseLinkClass_b190u9vd"
                href="https://www.algolia.com/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M229.3 182.6c-49.3 0-89.2 39.9-89.2 89.2 0 49.3 39.9 89.2 89.2 89.2s89.2-39.9 89.2-89.2c0-49.3-40-89.2-89.2-89.2zm62.7 56.6l-58.9 30.6c-1.8.9-3.8-.4-3.8-2.3V201c0-1.5 1.3-2.7 2.7-2.6 26.2 1 48.9 15.7 61.1 37.1.7 1.3.2 3-1.1 3.7zM389.1 32H58.9C26.4 32 0 58.4 0 90.9V421c0 32.6 26.4 59 58.9 59H389c32.6 0 58.9-26.4 58.9-58.9V90.9C448 58.4 421.6 32 389.1 32zm-202.6 84.7c0-10.8 8.7-19.5 19.5-19.5h45.3c10.8 0 19.5 8.7 19.5 19.5v15.4c0 1.8-1.7 3-3.3 2.5-12.3-3.4-25.1-5.1-38.1-5.1-13.5 0-26.7 1.8-39.4 5.5-1.7.5-3.4-.8-3.4-2.5v-15.8zm-84.4 37l9.2-9.2c7.6-7.6 19.9-7.6 27.5 0l7.7 7.7c1.1 1.1 1 3-.3 4-6.2 4.5-12.1 9.4-17.6 14.9-5.4 5.4-10.4 11.3-14.8 17.4-1 1.3-2.9 1.5-4 .3l-7.7-7.7c-7.6-7.5-7.6-19.8 0-27.4zm127.2 244.8c-70 0-126.6-56.7-126.6-126.6s56.7-126.6 126.6-126.6c70 0 126.6 56.6 126.6 126.6 0 69.8-56.7 126.6-126.6 126.6z"
                  />
                </svg>
              </a>
            </span>
            <span
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <a
                className="TechnologyLink_t1j6hkq4 baseLinkClass_b190u9vd"
                href="http://typescriptlang.org/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg>
                  __svg__
                </svg>
              </a>
            </span>
          </div>
        </div>
      </footer>
    </main>
  </div>,
]
`;
